build.gradle

```diff
@@ -188,19 +185,27 @@ task download(type: Copy) {
   into 'src/jar'
 }
 
-task gitRevList(type: Exec) {
+task gitRevList() {
   onlyIf {
     file('.git').exists()
   }
   inputs.dir('.git')
   outputs.files file('build/revision.txt')
-  commandLine 'git', 'rev-list', '--count', 'origin/main'
-  standardOutput = new ByteArrayOutputStream()
 
   doLast {
-    file('build/revision.txt').text = standardOutput.toString().trim()
+    def revision = grgit.log(includes:['HEAD']).size()
+    file('build/revision.txt').text = revision.toString().trim()
     // Update the version to the new revision
-    project.setVersion(standardOutput.toString().trim())
+    project.setVersion(revision.toString().trim())
+  }
+}
+
+task gitUpdate() {
+  doLast {
+  grgit.pull {
+    remote= 'origin'
+    branch= 'main'
+  }
   }
 }
 
```

Comment:
Should we keep svnRevList?

Reply:
> I don't see anything in here for detecting a modified repository and adding a 'M' to the version string. Did that concept go away? I found it useful.

I didn't do anything about that! It was always handled by the nemerosa versioning package.

Reply:
Deleted in the latest commit.